crates/sui-indexer-alt-graphql/src/api/types/available_range.rs

```diff
@@ -74,8 +75,58 @@ impl AvailableRange {
         Ok(Self {
             scope: scope.clone(),
             first,
+            pipelines,
         })
     }
+
+    // Get the lowest checkpoint for which data is available for the provided pipelines. Errors if
+    // no pipelines are found when instantiating the available range from the RententionKey.
+    pub(crate) fn reader_lo(&self) -> Result<u64, RpcError> {
+        if self.pipelines.is_empty() {
+            return Err(RpcError::InternalError(Arc::new(anyhow::anyhow!(
+                "At least one pipeline must be providedd to calculate the available range."
+            ))));
```

Comment:
This is something that you can implement as a test based off data generated by a macro (as discussed in the previous PR).